src/cache/messages.ml

```diff
@@ -72,10 +72,7 @@ let sexp_of_message : type a. version -> a message -> Sexp.t =
   | Dedup file ->
     cmd "dedup"
       [ Sexp.List
-          [ Sexp.Atom
-              (Path.Local.to_string
-                 (Path.Build.local file.in_the_build_directory))
-          ; Sexp.Atom (Path.to_string file.in_the_cache)
+          [ Sexp.Atom (Path.Local.to_string (Path.Build.local file.path))
```

Comment:
While the path simplification modification is a good one, we're breaking the protocol. The correct way to do this would be to bump the protocol version (to 1.3 I think) and still handle the old messages if v < 1.3, translating to the other semantic.

Reply:
Indeed, that's the right thing to do, and shouldn't be too difficult.